src/linux/mod.rs

```diff
@@ -85,3 +92,99 @@ impl Sandbox for LinuxSandbox {
         }
     }
 }
+
+/// Restrict networking using seccomp and namespaces.
+fn restrict_networking() -> Result<()> {
+    // Create network namespace.
+    let result = create_user_namespace(true).and_then(|_| unshare(Namespaces::NETWORK));
+
+    // Apply seccomp network filter.
+    match NetworkFilter::apply() {
+        _ if result.is_ok() => Ok(()),
+        result => result,
+    }
```

Comment:
This match statement is kinda hard to read... Mainly because it doesn't actually do any matching on the value... The logic here is just `Result::or`...

```suggestion
    result.or(NetworkFilter::apply())
```

Or maybe make it more obvious that `NetworkFilter::apply()` always runs using a variable:

```suggestion
    let seccomp_result = NetworkFilter::apply();

    result.or(seccomp_result)
```

Reply:
Not intentional, my clippy instincts just got the better of me.